base/linalg/cholesky.jl

```diff
@@ -84,14 +84,17 @@ function chol!{T}(A::AbstractMatrix{T}, uplo::Symbol)
     return uplo == :U ? UpperTriangular(A) : LowerTriangular(A)
 end
 
-function cholfact!{T<:BlasFloat}(A::StridedMatrix{T}, uplo::Symbol=:U; pivot=false, tol=0.0)
+cholfact!{T<:BlasFloat}(A::StridedMatrix{T}, uplo::Symbol=:U, pivot::Union(Type{Val{false}}, Type{Val{true}})=Val{false}; tol=0.0) =
+    _cholfact!(A, pivot, uplo, tol=tol)
+function _cholfact!{T<:BlasFloat}(A::StridedMatrix{T}, ::Type{Val{false}}, uplo::Symbol=:U; tol=0.0)
```

Comment:
Why are the underscore versions necessary? Couldn't the underscore versions just be used as the ordinary `cholfact!` methods instead of having `cholfact!` calling `_cholfact!`?